src/kv/tx_view.h

```diff
@@ -2,266 +2,83 @@
 // Licensed under the Apache 2.0 License.
 #pragma once
 
-#include "ds/champ_map.h"
+#include "kv/untyped_map.h"
+#include "kv/untyped_tx_view.h"
 #include "kv_types.h"
 
-#include <unordered_map>
-
 namespace kv
 {
-  static bool is_deleted(Version version)
-  {
-    return version < 0;
-  }
-
-  template <typename V>
-  struct VersionV
-  {
-    Version version;
-    V value;
-
-    VersionV() = default;
-    VersionV(Version ver, V val) : version(ver), value(val) {}
-  };
-
-  template <typename K, typename V, typename H>
-  using State = champ::Map<K, VersionV<V>, H>;
-
-  template <typename K, typename V, typename H>
-  using Read = std::unordered_map<K, Version, H>;
-
-  template <typename K, typename V, typename H = std::hash<K>>
-  using Write =
-    std::unordered_map<K, std::optional<V>, H>; //< nullopt indicates a deletion
-
-  // This is a container for a write-set + dependencies. It can be applied to a
-  // given state, or used to track a set of operations on a state
-  template <typename K, typename V, typename H = std::hash<K>>
-  struct ChangeSet
-  {
-  public:
-    using State = State<K, V, H>;
-    using Read = Read<K, V, H>;
-    using Write = Write<K, V, H>;
-
-    State state;
-    State committed;
-    Version start_version;
-
-    Version read_version = NoVersion;
-    Read reads = {};
-    Write writes = {};
-
-    ChangeSet(
-      State& current_state, State& committed_state, Version current_version) :
-      state(current_state),
-      committed(committed_state),
-      start_version(current_version)
-    {}
-
-    ChangeSet(ChangeSet&) = delete;
-  };
-
-  template <typename K, typename V, typename H = std::hash<K>>
-  class TxView
+  template <typename K, typename V, typename KSerialiser, typename VSerialiser>
```

Comment:
Useless diff. The left side is now `src/kv/untyped_tx_view.h`, minus the definitions that were pulled into `src/kv/change_set.h`. The right side was in `experimental.h`.